Readme: add explanation for new functions and cleanup some existing sections

TobiasKadelka opened the pull request:
Some functions got implemented that were not described in the readme until now or are completely new.
This PR adds an explanation about the new functions and subcommands, and corrects some things in the readme that were not up to date anymore.

aqw commented on README.md:
```diff
@@ -142,7 +145,8 @@ laptop asset should be converted to a folder, and the license moved into it.
   Initialize an Onyo repository. The directory will be initialized as a git
   repository (if it is not one already), the `.onyo/` configuration folder
-  created, and template files generated and committed.
+  created and default values will be set in `.onyo/config`, and template files
```
Proposed language:

> The directory will be initialized as a git repository (if it is not one already), the `.onyo/` directory created (containing default config files, templates, etc), and everything committed.

TobiasKadelka replied:
Yes, sounds much better.

aqw commented on README.md:
```diff
@@ -299,13 +303,46 @@ laptop asset should be converted to a folder, and the license moved into it.
+  After the editing is done, the new file will be checked for the validity of
+  it's YAML syntax.
+
+  - `--non-interactive` : Suppress opening of editor after file creation.
```
I'd like to clarify something here:
- `onyo new` will spawn an interactive prompt, and the user will enter information. It will then open an editor, for the remaining information to be entered.
- `onyo new --non-interactive` will spawn an interactive prompt, and the user will enter information. It will not open an editor.
- 
If this is the behavior, then at minimum `--non-interactive` is poorly named, as it still requires interaction.

To me, the utility of a feature like `--non-interactive` is to allow scripts to use it. How would one script a call to `onyo new` with sufficient information to not require interaction?

aqw commented on README.md:
```diff
@@ -299,13 +303,46 @@ laptop asset should be converted to a folder, and the license moved into it.
+- `onyo history [--non-interactive, -I] asset | directory`:
+
+  Show the history of a `directory` or `asset` file. The tools to show the
+  history are defined in `.onyo/config`.
```
Redundant. Remove here, It's said better 2-3 sentences later.

TobiasKadelka replied:
The sentence "The tools to show the  history are defined in `.onyo/config`" was removed.

aqw commented on README.md:
```diff
@@ -299,13 +303,46 @@ laptop asset should be converted to a folder, and the license moved into it.
+  with `onyo config history.interactive "git log --follow"`.
+- `onyo fsck`:
+
+  Checks if an `onyo repository` is given, which needs to be a valid git
```
I would like an opening more like: "Runs a comprehensive suite of checks to verify the integrity of the repository and contents of an onyo repository. Including: <list>"

TobiasKadelka replied:
Yes, sounds good. I started with that as a quick introduction sentence, before going into details.

aqw reviewed (changes requested):
Overall looks good! The primary issue is around `onyo new`. That one is unclear to me.

TobiasKadelka commented:
I cleaned up the Readme further to remove all information that was old or ideas that changed.

I also removed TODO sections, and added information about the goal of these sections to dedicated issues. When they are later defined/implemented they have to be documented in the correct form.
